### PR TITLE
Build using CMake

### PR DESCRIPTION
**Pull request by DesWurstes**
- Currently can't build tests. <-- WIP

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+set(LIBUNIVALUE_INTERFACE_AGE libunivalue_interface_age)
+
+# ABI version
+# http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+set(LIBUNIVALUE_CURRENT libunivalue_current)
+set(LIBUNIVALUE_REVISION libunivalue_revision)
+set(LIBUNIVALUE_AGE libunivalue_age)
+
+set(LIBTOOL_APP_LDFLAGS $ENV{LIBTOOL_APP_LDFLAGS})
+if (MINGW)
```
Comment (ken2812221) [inline CMakeLists.txt]: Why this is MINGW only?

**Comment by DesWurstes** [comment]
Thanks for the review! I have updated the patch.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+math(EXPR libunivalue_binary_age "100 * ${libunivalue_minor_version} + ${libunivalue_micro_version}")
+math(EXPR libunivalue_current "${libunivalue_binary_age} - ${libunivalue_interface_age}")
+set(libunivalue_age ${libunivalue_current})
+set(libunivalue_revision ${libunivalue_interface_age})
+set(libunivalue_version "${libunivalue_major_version}.${libunivalue_minor_version}.${libunivalue_micro_version}${libunivalue_extraversion}")
+
+project(libunivalue
+    VERSION "${libunivalue_major_version}.${libunivalue_minor_version}.${libunivalue_micro_version}"
+    HOMEPAGE_URL "http://github.com/jgarzik/univalue/"
+    LANGUAGES CXX)
```
Comment (ken2812221) [inline CMakeLists.txt]: HOMEPAGE_URL need the latest cmake 3.12. https://cmake.org/cmake/help/v3.5/command/project.html. I think you can just delete that line.

**Comment by ken2812221** [comment]
Tested the follwing sets ACK f8a2a06
- `cmake -G "Unix Makefiles"`
- `cmake -G "Visual Studio 15 2017 Win64"`
- `cmake -G "Visual Studio 15 2017"`

**Pull request closed without merging**